Realm/Realm.Weaver/RealmWeaver.cs

```diff
@@ -381,13 +381,21 @@ private WeavePropertyResult WeaveProperty(PropertyDefinition prop, TypeDefinitio
                     return WeavePropertyResult.Warning($"{type.Name}.{prop.Name} is not an automatic property but its type is a RealmObject/EmbeddedObject which normally indicates a relationship.");
                 }
 
+                if (prop.ContainsAsymmetricObject(_references))
+                {
+                    return WeavePropertyResult.Warning($"{type.Name}.{prop.Name} is not an automatic property but its type is a AsymmetricObject. This usually indicates a relationship but AsymmetricObjects are not allowed to be the receiving end of any relationships.");
+                }
+
                 return WeavePropertyResult.Skipped();
             }
 
             var backlinkAttribute = prop.CustomAttributes.FirstOrDefault(a => a.AttributeType.Name == "BacklinkAttribute");
-            if (backlinkAttribute != null && !prop.IsIQueryable())
+            if (backlinkAttribute != null)
             {
-                return WeavePropertyResult.Error($"{type.Name}.{prop.Name} has [Backlink] applied, but is not IQueryable.");
+                if (!prop.IsIQueryable())
+                {
+                    return WeavePropertyResult.Error($"{type.Name}.{prop.Name} has [Backlink] applied, but is not IQueryable.");
+                }
```

Comment:
Why was that split?

Reply:
A forgotten leftover of a check that later was moved down. Thank you for noticing it.